src/objects/GameObject.cpp

```diff
@@ -6,9 +6,10 @@
 #include "core/API.h"
 
 GameObject::GameObject(sc2::Tag tag_): m_tag(tag_) {
+    m_unit = gAPI->observer().GetUnit(tag_);
 }
 
-GameObject::GameObject(const sc2::Unit& unit_): m_tag(unit_.tag) {
+GameObject::GameObject(const sc2::Unit& unit_): m_unit(&unit_), m_tag(unit_.tag) {
```

Comment:
Since we are going to keep sc2::Unit inside please pass it by pointer. Please, check the linter in advance.